server/src/main/java/org/elasticsearch/index/engine/ReadOnlyEngine.java

```diff
@@ -0,0 +1,349 @@
+        return 0;
+    }
+
+    @Override
+    public boolean hasCompleteOperationHistory(String source, MapperService mapperService, long startingSeqNo) throws IOException {
+        return false;
+    }
+
+    @Override
+    public TranslogStats getTranslogStats() {
+        return translogStats;
+    }
+
+    @Override
+    public Translog.Location getTranslogLastWriteLocation() {
+        return new Translog.Location(0,0,0);
+    }
+
+    @Override
+    public long getLocalCheckpoint() {
+        return seqNoStats.getLocalCheckpoint();
+    }
+
+    @Override
+    public void waitForOpsToComplete(long seqNo) { }
+
+    @Override
+    public void resetLocalCheckpoint(long newCheckpoint) {
+    }
+
+    @Override
+    public SeqNoStats getSeqNoStats(long globalCheckpoint) {
+        return new SeqNoStats(seqNoStats.getMaxSeqNo(), seqNoStats.getLocalCheckpoint(), globalCheckpoint);
+    }
+
+    @Override
+    public long getLastSyncedGlobalCheckpoint() {
+        return seqNoStats.getGlobalCheckpoint();
+    }
+
+    @Override
+    public long getIndexBufferRAMBytesUsed() {
+        return 0;
+    }
+
+    @Override
+    public List<Segment> segments(boolean verbose) {
+        return Arrays.asList(getSegmentInfo(lastCommittedSegmentInfos, verbose));
+    }
+
+    @Override
+    public void refresh(String source) throws EngineException {}
+
+    @Override
+    public void writeIndexingBuffer() throws EngineException {}
+
+    @Override
+    public boolean shouldPeriodicallyFlush() {
+        return false;
+    }
+
+    @Override
+    public SyncedFlushResult syncFlush(String syncId, CommitId expectedCommitId) throws EngineException {
+        // we can't do synced flushes this would require an indexWriter which we don't have
+        return SyncedFlushResult.COMMIT_MISMATCH;
+    }
+
+    @Override
+    public CommitId flush(boolean force, boolean waitIfOngoing) throws EngineException {
+        return new CommitId(lastCommittedSegmentInfos.getId());
+    }
+
+    @Override
+    public void forceMerge(boolean flush, int maxNumSegments, boolean onlyExpungeDeletes,
+                           boolean upgrade, boolean upgradeOnlyAncientSegments) { }
+
+    @Override
+    public IndexCommitRef acquireLastIndexCommit(boolean flushFirst) {
+        store.incRef();
+        return new IndexCommitRef(indexCommit, store::decRef);
+    }
+
+    @Override
+    public IndexCommitRef acquireSafeIndexCommit() {
+        return acquireLastIndexCommit(false);
+    }
+
+    @Override
+    public void activateThrottling() {
+    }
+
+    @Override
+    public void deactivateThrottling() {
+    }
+
+    @Override
+    public void trimUnreferencedTranslogFiles() { }
+
+    @Override
+    public boolean shouldRollTranslogGeneration() {
+        return false;
+    }
+
+    @Override
+    public void rollTranslogGeneration() throws EngineException { throw new UnsupportedOperationException(); }
```

Comment:
new lines missing. Also we should assert this is never called. Also no need for `throws EngineException`